---
title: "Combined RNAi screen"
author: "Nathaniel Evans"
date: "October 4, 2019"
output:
  html_document: 
    theme: flatly 
    highlight: kate
    toc: true
    toc_depth: 2
    toc_float : true
    smooth_scroll: true
    number_sections : false
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(data.table)
library(VennDiagram)
library(PharmacoGx)

```


# overview 






# Data

## Cell Line Metadata 

DepMap Cell Line metadata 

[source](https://ndownloader.figshare.com/files/16757723)

```{r}
meta.dat <- as.data.frame(fread('./../data/sample_info.csv', sep=','))
meta.dat %>% head()
```

## Combined RNAi

[more info](https://depmap.org/R2-D2/)  

[source](https://ndownloader.figshare.com/files/13515395)  

#### References 

James M. McFarland, Zandra V. Ho, Guillaume Kugener, Joshua M. Dempster, Phillip G. Montgomery, Jordan G. Bryan, John M. Krill-Burger, Thomas M. Green, Francisca Vazquez, Jesse S. Boehm, Todd R. Golub, William C. Hahn, David E. Root, Aviad Tsherniak. (2018). Improved estimation of cancer dependencies from large-scale RNAi screens using model-based normalization and data integration. Nature Communications 9, 1. https://doi.org/10.1038/s41467-018-06916-5 


```{r}
dep.RNAi <- as.data.frame(fread('./../data/D2_combined_gene_dep_scores.csv', sep=',')) %>% pivot_longer(-V1, names_to='cell_line') %>% mutate(gene = V1, effect = value) %>% select(-V1, -value) %>% merge(meta.dat, by.x='cell_line', by.y='CCLE Name', all.x=T)
#dep.RNAi %>% head()
 
rnai.dat <- dep.RNAi %>% select(cell_line, gene, effect, DepMap_ID) %>% mutate(response=effect, target=gene, response_type='RNAi_dependency') %>% select(DepMap_ID, target, response, response_type) 
rnai.dat %>% head()
```


## CRISPR (Avana)

[more info](https://depmap.org/portal/download/)

[source](https://ndownloader.figshare.com/files/16757666)

### 'omics datasets:
Mahmoud Ghandi, Franklin W. Huang, Judit Jané-Valbuena, Gregory V. Kryukov, ... Todd R. Golub, Levi A. Garraway & William R. Sellers. 2019. Next-generation characterization of the Cancer Cell Line Encyclopedia. Nature 569, 503-508 (2019).

#### References 

DepMap, Broad (2019): DepMap 19Q3 Public. figshare. Dataset doi:10.6084/m9.figshare.9201770.v1.

Robin M. Meyers, Jordan G. Bryan, James M. McFarland, Barbara A. Weir, ... David E. Root, William C. Hahn, Aviad Tsherniak. Computational correction of copy number effect improves specificity of CRISPR-Cas9 essentiality screens in cancer cells. Nature Genetics 2017 October 49:1779-1784. doi:10.1038/ng.3984

```{r}
dep.CRISPR <- as.data.frame(fread('./../data/Achilles_gene_effect.csv', sep=',')) %>% pivot_longer(-V1, names_to='gene') %>% mutate(cell_line = V1, dep = value) %>% select(-V1, -value) %>% merge(meta.dat, by.x='cell_line', by.y='DepMap_ID', all.x=T)
#dep.CRISPR %>% head()

crispr.dat <- dep.CRISPR %>% select(cell_line, gene, dep) %>% mutate(DepMap_ID=cell_line, target=gene, response=dep, response_type='crispr_dependency') %>% select(DepMap_ID,target,response,response_type)
crispr.dat %>% head()
```

## CCLE Expression 

[Cellular Models Expression Public 19Q3](https://depmap.org/portal/download/)

[source](https://ndownloader.figshare.com/files/16757690)

# References 

Mahmoud Ghandi, Franklin W. Huang, Judit Jané-Valbuena, Gregory V. Kryukov, ... Todd R. Golub, Levi A. Garraway & William R. Sellers. 2019. Next-generation characterization of the Cancer Cell Line Encyclopedia. Nature 569, 503-508 (2019).

```{r}
CCLE.rnaSeq <- as.data.frame(fread('./../data/CCLE_expression.csv', sep=',')) %>% pivot_longer(-V1, names_to='gene') %>% mutate(cell_line = V1, expression = value) %>% select(-V1, -value)

CCLE.rnaSeq %>% head()
```

## Drug Response (PRISM)

Pooled cell line drug response data; measured by fold change of cell line abundance. 

[Drug sensitivity (PRISM Repurposing Primary Screen) 19Q3](https://depmap.org/portal/download/)

[source](https://ndownloader.figshare.com/files/17008628)

#### References 

Steven M Corsello, Rohith T Nagari, Ryan D Spangler, Jordan Rossen, Mustafa Kocak, Jordan G Bryan, Ranad Humeidi, David Peck, Xiaoyun Wu, Andrew A Tang, Vickie MWang, Samantha A Bender, Evan Lemire, Rajiv Narayan, Philip Montgomery, Uri Ben-David, Yejia Chen, Matthew G Rees, Nicholas J Lyons, James M McFarland, Bang TWong, Li Wang, Nancy Dumont, Patrick J O'Hearn, Eric Stefan, John G Doench, HeidiGreulich, Matthew Meyerson, Francisca Vazquez, Aravind Subramanian, Jennifer A Roth, Joshua A Bittker, Jesse S Boehm, Christopher C Mader, Aviad Tsherniak, Todd R Golub. 2019. Non-oncology drugs are a source of previously unappreciated anti-cancer activity. bioRxiv doi: 10.1101/730119


```{r}
drug.PRISM <- as.data.frame(fread('./../data/primary_replicate_collapsed_logfold_change.csv', sep=',')) %>% pivot_longer(-V1, names_to='drug') %>% mutate(cell_line = V1, CV_fold_change = value) %>% select(-V1, -value)
#drug.PRISM %>% head()

PRISM.meta <- as.data.frame(fread('./../data/primary_replicate_collapsed_treatment_info.csv', sep=','))
#PRISM.meta %>% head()

PRISM.dat <- drug.PRISM %>% merge(unique(select(PRISM.meta, column_name, name, target)), by.x='drug', by.y='column_name', all.x=T) %>% mutate(drug_name=name) %>% select(-name) %>% mutate(DepMap_ID=cell_line, response=CV_fold_change, response_type='pooled_drugresp_prism') %>% select(DepMap_ID, drug_name, target, response, response_type)

PRISM.dat %>% head()
```

## GDSC Expression 

[more info](https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-3610/) 

There is no good bulk download page, so we'll use PharmcoGx 

I DON'T KNOW HOW TO GET GDSC CELL LINES AHHHHHHHHHHHHH

```{r}
data("GDSC")

GDSC.expr <- summarizeMolecularProfiles(common$GDSC, cellNames(common$GDSC), mDataType="rna", features=commonGenes, verbose=FALSE)

GDSC.expr[1:5, 1:5]
```


## Drug Response (GDSC1)

Genomics of Drug Sensitivity in Cancer (Sanger)

[data dict](ftp://ftp.sanger.ac.uk/pub/project/cancerrxgene/releases/current_release/GDSC_Fitted_Data_Description.pdf)
[source-data](ftp://ftp.sanger.ac.uk/pub/project/cancerrxgene/releases/current_release/GDSC1_fitted_dose_response_15Oct19.xlsx)

[source-anova](ftp://ftp.sanger.ac.uk/pub/project/cancerrxgene/releases/current_release/ANOVA_results_GDSC1_15Oct19.xlsx)
[source-anova info](https://www.cancerrxgene.org/downloads/bulk_download)

#### References

Genomics of Drug Sensitivity in Cancer (GDSC): a resource for therapeutic biomarker discovery in cancer cells.
Yang et al., (2013) Nucl. Acids Res. 41 (Database issue): D955 - D961. (PMID:23180760  )

A landscape of pharmacogenomic interactions in cancer
Iorio et al., (2016). Cell, Volume 166, Issue 3, 740 - 754 (PMID:27397505 )

Systematic identification of genomic markers of drug sensitivity in cancer cells
Garnett et al., (2012) Nature volume 483, pages 570 - 575 (PMID:27397505 )

```{r}

GDSC.targets <- readxl::read_excel('./../data/ANOVA_results_GDSC1_15Oct19.xlsx')
#GDSC.targets %>% head()

#drug.GDSC %>% head()

drug.GDSC <- readxl::read_excel('./../data/GDSC1_fitted_dose_response_15Oct19.xlsx')
#drug.GDSC <- GDSC.targets %>% select(drug_name, target) %>% unique() %>% merge(drug.GDSC, ., by.x='drug', by.y = 'drug_name', all.x=T)

GDSC.dat <- drug.GDSC %>% select(CELL_LINE_NAME, DRUG_NAME, PUTATIVE_TARGET, AUC)
GDSC.dat %>% head()
```


## Drug Response (CTRP CTD)

NOT SURE WHERE THE CELL LINE NAME / DATA IS 

CTRP CTD^2

[source](ftp://caftpd.nci.nih.gov/pub/OCG-DCC/CTD2/Broad/CTRPv2.0_2015_ctd2_ExpandedDataset/CTRPv2.0_2015_ctd2_ExpandedDataset.zip)

#### References 

Harnessing Connectivity in a Large-Scale Small-Molecule Sensitivity Dataset Seashore-Ludlow et al., Cancer Discovery, 5, 1210-1223 (2015) and Correlating chemical sensitivity and basal gene expression reveals mechanism of action Rees et al., Nat Chem Biol, 12, 109-116 (2016)

```{r}
# use master_cpd_id to merge to drug name 
CTRP.compound.meta <- as.data.frame(fread('./../data/CTRPv2.0_2015_ctd2_ExpandedDataset/v20.meta.per_compound.txt', sep='\t'))

#CTRP.compound.meta %>% head() 

CTRP.cell.line.meta <- as.data.frame(fread('./../data/CTRPv2.0_2015_ctd2_ExpandedDataset/v20.meta.per_cell_line.txt', sep='\t'))

CTRP.cell.line.meta %>% head()

CTRP.exp.meta <- as.data.frame(fread('./../data/CTRPv2.0_2015_ctd2_ExpandedDataset/v20.meta.per_experiment.txt', sep='\t')) # this has the link we need

CTRP.exp.meta %>% head()

drug.CTRP <- as.data.frame(fread('./../data/CTRPv2.0_2015_ctd2_ExpandedDataset/v20.data.curves_post_qc.txt', sep='\t')) %>% select(experiment_id, master_cpd_id, area_under_curve) %>% merge(select(CTRP.compound.meta, master_cpd_id, cpd_name, gene_symbol_of_protein_target, target_or_activity_of_compound), by='master_cpd_id') %>% merge(select(CTRP.exp.meta, experiment_id, master_ccl_id), by='experiment_id') %>% merge(select(CTRP.cell.line.meta, master_ccl_id, ccl_name, ccle_primary_site), by='master_ccl_id') 

drug.CTRP %>% head()

CTRP.dat <- drug.CTRP %>% mutate(response=area_under_curve, response_type='AUC_drugresp_CTRP', drug_name=cpd_name,target=gene_symbol_of_protein_target) %>% select(drug_name, target, response, response_type, ccl_name, ccle_primary_site)
CTRP.dat %>% head()

```



## CCLE NP24 

Pharmacologic profiles for 24 anticancer drugs across 504 cell lines.

[more info](https://depmap.org/portal/ccle/)

[source- drug data](https://depmap.org/portal/download/api/download/external?file_name=ccle_legacy_data%2Fpharmacological_profiling%2FCCLE_NP24.2009_Drug_data_2015.02.24.csv)

[source- profile data](https://depmap.org/portal/download/api/download/external?file_name=ccle_legacy_data%2Fpharmacological_profiling%2FCCLE_NP24.2009_profiling_2012.02.20.csv)


[source- activity data](https://depmap.org/portal/download/api/download/external?file_name=ccle_legacy_data%2Fpharmacological_profiling%2FCCLE_GNF_data_090613.xls)

### Act Area = AUC Area 

> Additionally, we calculated two forms of the Activity area for each curve,
defined as the area between the response curve and a fixed reference Aref = 0 or a variable
reference Aref = max(0, Alow) where Alow is the activity at the lowest concentration, up to
the maximum tested concentration. In practice, the Activity area was calculated as the
sum of differences between the measured Ai at concentration i and the reference level.
Thus, using the fixed reference, Activity area = 0 corresponds to an inactive compound,
and 8 corresponds to a compound which had A = -100% at all eight concentrations
points. The variable reference form was introduced to adjust for curves with large
positive activities close to zero concentration, which are usually artifacts of imperfectly
corrected variations on the assay plate. For this measure, the median of all replicate
activity values was used regardless of cell line run day. To prevent confusion, the Activity
Area was calculated using Aref = 0 unless otherwise noted. [supplementary, pg11](https://media.nature.com/original/nature-assets/nature/journal/v483/n7391/extref/nature11003-s2.pdf)  


#### References 

Cancer Cell Line Encyclopedia Consortium, and Genomics of Drug Sensitivity in Cancer Consortium. 2015. Pharmacogenomic Agreement between Two Cancer Cell Line Data Sets. Nature 528 (7580):84-87. https://doi.org/10.1038/nature15736.

Jordi Barretina, Giordano Caponigro, Nicolas Stransky, Kavitha Venkatesan, William R. Sellers, Robert Schlegel, Levi A. Garraway, et. al. 2012. The Cancer Cell Line Encyclopedia Enables Predictive Modelling of Anticancer Drug Sensitivity. Nature 483 (7391):603-7. https://doi.org/10.1038/nature11003.

```{r}
drug.CCLE <- as.data.frame(fread('./../data/CCLE_NP24.2009_Drug_data_2015.02.24.csv', sep=','))
#drug.CCLE %>% head()

CCLE.profile <- as.data.frame(fread('./../data/CCLE_NP24.2009_profiling_2012.02.20.csv', sep=','))
#CCLE.profile %>% head()

CCLE.activity <-readxl::read_excel('./../data/CCLE_GNF_data_090613.xls')
#CCLE.activity %>% head()

# relevant features 
CCLE.dat <- drug.CCLE %>% select(`CCLE Cell Line Name`, Compound, Target, ActArea) %>% merge(unique(select(meta.dat, DepMap_ID, `CCLE Name`)), by.x='CCLE Cell Line Name', by.y='CCLE Name') %>% mutate(response=ActArea, target=Target, drug_name=Compound, response_type='AUC_drug_CCLE') %>% select(DepMap_ID,drug_name, target, response, response_type)
CCLE.dat %>% head()
```

## Targetome 

--- 
### From Rory 
---

#### targetome_DrugInformation_070617.txt
- drugs by class/categories (ATC, EPC, etc) + FDA approval date

#### targetome_DrugSynonymDictionary_100917.txt
- mapping for drug->synonyms
- includes chemical, brand, generic, proprietary, investigational names as available
- note IMO it's best to use the full synonym dictionary because there is poor consistency about drug name usage :/

#### targetome_Evidence_TIPS_101017.txt
- includes target evidence for ~140 drugs

#### targetome_FullEvidence_011019.txt
- includes target evidence for ~170 drugs
- this is all drugs we originally mined, note we subsetted to anti-neoplastic (ATC class with prefix "L01" for cancer targetome paper)
- +small bug fix, there were a few rows with blank assay values due to a range of nM values being reported, added "range" flag to indicate

---

Drug target information

[more info](https://www.ncbi.nlm.nih.gov/pubmed/28964549)

#### References 

Evidence-Based Precision Oncology with the Cancer Targetome. - PubMed - NCBI. https://www.ncbi.nlm.nih.gov/pubmed/28964549.

```{r}

targetome.druginfo <- as.data.frame(fread('./../data/targetome/Targetome_DrugInformation_070617.txt', sep='\t'))
targetome.drugsynonym <- as.data.frame(fread('./../data/targetome/Targetome_DrugSynonymDictionary_100917.txt', sep='\t'))
targetome.evidence.tips <- as.data.frame(fread('./../data/targetome/Targetome_Evidence_TIPS_101017.txt', sep='\t'))
targetome.evidence.full <- as.data.frame(fread('./../data/targetome/Targetome_FullEvidence_011019.txt', sep='\t'))

targetome.druginfo %>% head() 
targetome.drugsynonym %>% head()
targetome.evidence.tips %>% head()
targetome.evidence.full %>% head()

```

# EDA 

## CCLE Cell-Line Overlap

Does our CCLE gene expression have all the cell lines used in the gene dependency datasets? 

```{r}

cell.lines.exp <- CCLE.rnaSeq %>% select(cell_line) %>% unique() %>% drop_na() %>% arrange(cell_line) %>% .$cell_line

cell.lines.rnai <- rnai.dat %>% select(DepMap_ID) %>% unique() %>% drop_na() %>% .$DepMap_ID

cell.lines.crispr <- crispr.dat %>% select(cell_line) %>% unique() %>% drop_na() %>% .$cell_line

cell.lines.CCLE <- CCLE.dat %>% select(DepMap_ID) %>% unique() %>% drop_na() %>% arrange(DepMap_ID) %>% .$`DepMap_ID`

cell.lines.GDSC <- GDSC.dat %>% select(CELL_LINE_NAME) %>% unique() %>% drop_na() %>% arrange(CELL_LINE_NAME) %>% .$CELL_LINE_NAME

venn.diagram(
  x = list(cell.lines.exp, cell.lines.rnai, cell.lines.crispr, cell.lines.CCLE),
  category.names = c("CCLE EXPR" , "RNAi DEP" , "CRISPR DEP", "CCLE DRUG"),
  filename = './figs/cell_line_venn.png',
  output=T
)

venn.diagram(
  x = list(cell.lines.exp, cell.lines.CCLE, cell.lines.GDSC),
  category.names = c("CCLE EXPR", "CCLE DRUG", "GDSC DRUG"),
  filename = './figs/cell_line_venn2.png',
  output=T
)

```

Most of the cell lines from the dependency scores are in CCLE.EXPR
![](./figs/cell_line_venn.png)


FOR some reason the CCLE Drug data cell lines are not in the CCLE Expr dataset.  
![](./figs/cell_line_venn2.png)

```{r}
cell.lines.exp[1:25]
cell.lines.CCLE[1:25]
cell.lines.GDSC[1:25]
```

## Gene Dependency Coverage 

How many of the gene dependency assays overlap between the RNAi and CRISPR screens in our RNA expression? 

```{r}
genes.exp <- CCLE.rnaSeq %>% select(gene) %>% unique() %>% drop_na() %>% .$gene

genes.rnai <- rnai.dat %>% select(gene) %>% unique() %>% drop_na() %>% .$gene

genes.crispr <- crispr.dat %>% select(gene) %>% unique() %>% drop_na() %>% .$gene

venn.diagram(
  x = list(genes.exp, genes.rnai, genes.crispr),
  category.names = c("EXPR GENES" , "RNAi GENES" , "CRISPR GENES"),
  filename = './figs/gene_venn.png',
  output=T
)

```

![](./figs/gene_venn.png)







