---
title: "Combined RNAi screen"
author: "Nathaniel Evans"
date: "October 4, 2019"
output:
  html_document: 
    theme: flatly 
    highlight: kate
    toc: true
    toc_depth: 2
    toc_float : true
    smooth_scroll: true
    number_sections : false
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyr)
library(ggplot2)
library(data.table)
library(VennDiagram)
library(readr)

```

# overview 

The purpose of this script is to read in the various datatypes, document sources, document any necessary normalization. 

# DepMap Cell Line Metadata 

This data set links `DepMap_ID` to various other genomic identrifiers.  

[source](https://ndownloader.figshare.com/files/16757723)

```{r}
meta.dat <- as.data.frame(fread('./../data/sample_info.csv', sep=','))
meta.dat %>% head()

DepMap.cosmic.ids <- meta.dat %>% select(COSMIC_ID) %>% drop_na() %>% .$COSMIC_ID

```

# Combined RNAi

## source  

[more info](https://depmap.org/R2-D2/)  

[source](https://ndownloader.figshare.com/files/13515395)  

## References 

James M. McFarland, Zandra V. Ho, Guillaume Kugener, Joshua M. Dempster, Phillip G. Montgomery, Jordan G. Bryan, John M. Krill-Burger, Thomas M. Green, Francisca Vazquez, Jesse S. Boehm, Todd R. Golub, William C. Hahn, David E. Root, Aviad Tsherniak. (2018). Improved estimation of cancer dependencies from large-scale RNAi screens using model-based normalization and data integration. Nature Communications 9, 1. https://doi.org/10.1038/s41467-018-06916-5 


```{r}
dep.RNAi <- as.data.frame(fread('./../data/D2_combined_gene_dep_scores.csv', sep=',')) %>% pivot_longer(-V1, names_to='cell_line') %>% mutate(gene = V1, effect = value) %>% select(-V1, -value) %>% merge(meta.dat, by.x='cell_line', by.y='CCLE Name', all.x=T)
#dep.RNAi %>% head()
 
rnai.dat <- dep.RNAi %>% select(cell_line, gene, effect, DepMap_ID) %>% mutate(drug_name='rnai screen',response=effect, target=gene, response_type='RNAi_dependency') %>% select(drug_name, DepMap_ID, target, response, response_type) %>% mutate(id=DepMap_ID, id_type='DepMap_ID') %>% drop_na()  %>% select(-DepMap_ID)
rnai.dat %>% head()
```

# CRISPR (Avana)

## Sources 

[more info](https://depmap.org/portal/download/)

[source](https://ndownloader.figshare.com/files/16757666)

## References 

DepMap, Broad (2019): DepMap 19Q3 Public. figshare. Dataset doi:10.6084/m9.figshare.9201770.v1.

Robin M. Meyers, Jordan G. Bryan, James M. McFarland, Barbara A. Weir, ... David E. Root, William C. Hahn, Aviad Tsherniak. Computational correction of copy number effect improves specificity of CRISPR-Cas9 essentiality screens in cancer cells. Nature Genetics 2017 October 49:1779-1784. doi:10.1038/ng.3984

### 'omics datasets:
Mahmoud Ghandi, Franklin W. Huang, Judit Jané-Valbuena, Gregory V. Kryukov, ... Todd R. Golub, Levi A. Garraway & William R. Sellers. 2019. Next-generation characterization of the Cancer Cell Line Encyclopedia. Nature 569, 503-508 (2019).

```{r}
dep.CRISPR <- as.data.frame(fread('./../data/Achilles_gene_effect.csv', sep=',')) %>% pivot_longer(-V1, names_to='gene') %>% mutate(cell_line = V1, dep = value) %>% select(-V1, -value) %>% merge(meta.dat, by.x='cell_line', by.y='DepMap_ID', all.x=T)
#dep.CRISPR %>% head()

crispr.dat <- dep.CRISPR %>% select(cell_line, gene, dep) %>% mutate(drug_name='crispr_screen', DepMap_ID=cell_line, target=gene, response=dep, response_type='crispr_dependency') %>% select(drug_name, DepMap_ID,target,response,response_type) %>% mutate(id=DepMap_ID, id_type='DepMap_ID') %>% drop_na() %>% select(-DepMap_ID)

crispr.dat %>% head()
```

# CCLE Expression 

This is the expression data we will use for all datasets except beatAML. 

---
### From Ghandi et. al. 

> **RNA-seq profiling:** RNA-seq and analysis were performed for 1,019 cell lines as previously described5. In summary, non-strand-specific RNA sequencing was performed using large-scale, automated method of the Illumina TruSeq RNA Sample Preparation protocol. Oligo-dT beads were used to select polyadenylated mRNA. The selected RNA was then heat fragmented and randomly primed before cDNA synthesis. To maximize power to detect fusions, the insert size of fragments was set to 400 nt. The resultant cDNA then went through Illumina library preparation (end-repair, base 'A' addition, adaptor ligation, and enrichment) using Broad-designed indexed adapters for multiplexing. Sequencing was performed on the Illumina HiSeq 2000 or HiSeq 2500 instruments with sequence coverage of no less than 100 million paired 101 nucleotides-long reads per sample.

> **Short read alignment and calculation of gene *expression**: RNA-seq reads were aligned to the GRCh37 build of the human genome reference using STAR 2.4.2a59. The GENCODE v19 annotation was used for the STAR alignment and all other quantifications. Gene level RPKM and read count values were calculated using RNA-SeQC v1.1.860. Exon-exon junction read counts were obtained from STAR. Isoform-level expression in TPM (transcripts per million) was quantified using RSEM v.1.2.22. All methods were run as part of the pipeline developed for the GTEx Consortium (https://gtexportal.org)61.

---
### From DepMap Description (download page)

File Description
RNAseq **TPM** gene expression data for just protein coding genes using RSEM. Log2 transformed, using a pseudo-count of 1.

Rows: cell lines (Broad IDs)
Columns: genes (HGNC symbol and Entrez ID)

---

But What type of normalization? [TPM,RPKM]
-> This expression dataset uses **TPM** for **protein coding genes** and is **log2 transformed** using a pseudo-count of 1. They use [RSEM](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-323#Sec17) for this normalization. 

[RSEM tutorial](https://github.com/bli25broad/RSEM_tutorial)

## Sources

[Cellular Models Expression Public 19Q3](https://depmap.org/portal/download/)

[source](https://ndownloader.figshare.com/files/16757690)

## References 

Mahmoud Ghandi, Franklin W. Huang, Judit Jané-Valbuena, Gregory V. Kryukov, ... Todd R. Golub, Levi A. Garraway & William R. Sellers. 2019. Next-generation characterization of the Cancer Cell Line Encyclopedia. Nature 569, 503-508 (2019).

```{r}
map <- data.frame(fread('../data/processed/entrez_to_ensembl_mapping.csv'))

CCLE.rnaSeq <- as.data.frame(fread('./../data/CCLE_expression.csv', sep=',')) %>% pivot_longer(-V1, names_to='gene') %>% mutate(cell_line = V1, expression = value) %>% select(-V1, -value) %>% separate(gene, into = c("HGNC_sym", "entrez_id"), sep = " \\(") %>% mutate(entrez_id=substr(entrez_id, 1, nchar(entrez_id)-1)) %>% merge(map, by='entrez_id', all.x=T)

CCLE.rnaSeq %>% head()
```

```{r}
#map <- data.frame(fread('../data/processed/entrez_to_ensembl_mapping.csv'))

map2 <- CCLE.rnaSeq %>% select(HGNC_sym, entrez_id) %>% unique() %>% merge(map, by='entrez_id', all.x=T, all.y=F) 

map2 %>% head()
```

```{r}
write.csv(map2, '../data/processed/gene_id_map.csv', row.names=F)
```

```{r}
CCLE.rnaSeq %>% write_delim('./../data/processed/depmap_expression.csv', delim=',')
```

# Pooled Drug Response (PRISM)

Pooled cell line drug response data; measured by fold change of cell line abundance. 

## Sources 

[Drug sensitivity (PRISM Repurposing Primary Screen) 19Q3](https://depmap.org/portal/download/)

[source-data](https://ndownloader.figshare.com/files/17008628)
[source-info](https://ndownloader.figshare.com/files/17008634)

## References 

Steven M Corsello, Rohith T Nagari, Ryan D Spangler, Jordan Rossen, Mustafa Kocak, Jordan G Bryan, Ranad Humeidi, David Peck, Xiaoyun Wu, Andrew A Tang, Vickie MWang, Samantha A Bender, Evan Lemire, Rajiv Narayan, Philip Montgomery, Uri Ben-David, Yejia Chen, Matthew G Rees, Nicholas J Lyons, James M McFarland, Bang TWong, Li Wang, Nancy Dumont, Patrick J O'Hearn, Eric Stefan, John G Doench, HeidiGreulich, Matthew Meyerson, Francisca Vazquez, Aravind Subramanian, Jennifer A Roth, Joshua A Bittker, Jesse S Boehm, Christopher C Mader, Aviad Tsherniak, Todd R Golub. 2019. Non-oncology drugs are a source of previously unappreciated anti-cancer activity. bioRxiv doi: 10.1101/730119

```{r}
drug.PRISM <- as.data.frame(fread('./../data/primary_replicate_collapsed_logfold_change.csv', sep=',')) %>% pivot_longer(-V1, names_to='drug') %>% mutate(cell_line = V1, CV_fold_change = value) %>% select(-V1, -value)
#drug.PRISM %>% head()

PRISM.meta <- as.data.frame(fread('./../data/primary_replicate_collapsed_treatment_info.csv', sep=','))
#PRISM.meta %>% head()

PRISM.dat <- drug.PRISM %>% merge(unique(select(PRISM.meta, column_name, name, target)), by.x='drug', by.y='column_name', all.x=T) %>% mutate(drug_name=name) %>% select(-name) %>% mutate(DepMap_ID=cell_line, response=CV_fold_change, response_type='pooled_drugresp_prism') %>% select(DepMap_ID, drug_name, target, response, response_type) %>% mutate(id=DepMap_ID, id_type='DepMap_ID') %>% drop_na()  %>% select(-DepMap_ID)

PRISM.dat %>% head()
```


# Drug Response (GDSC1)

Genomics of Drug Sensitivity in Cancer (Sanger)

## Sources 

[data dict](ftp://ftp.sanger.ac.uk/pub/project/cancerrxgene/releases/current_release/GDSC_Fitted_Data_Description.pdf)
[source-data](ftp://ftp.sanger.ac.uk/pub/project/cancerrxgene/releases/current_release/GDSC1_fitted_dose_response_15Oct19.xlsx)

[source-anova](ftp://ftp.sanger.ac.uk/pub/project/cancerrxgene/releases/current_release/ANOVA_results_GDSC1_15Oct19.xlsx)
[source-anova info](https://www.cancerrxgene.org/downloads/bulk_download)

## References

Genomics of Drug Sensitivity in Cancer (GDSC): a resource for therapeutic biomarker discovery in cancer cells.
Yang et al., (2013) Nucl. Acids Res. 41 (Database issue): D955 - D961. (PMID:23180760  )

A landscape of pharmacogenomic interactions in cancer
Iorio et al., (2016). Cell, Volume 166, Issue 3, 740 - 754 (PMID:27397505 )

Systematic identification of genomic markers of drug sensitivity in cancer cells
Garnett et al., (2012) Nature volume 483, pages 570 - 575 (PMID:27397505 )

```{r}

GDSC.targets <- readxl::read_excel('./../data/ANOVA_results_GDSC1_15Oct19.xlsx')
#GDSC.targets %>% head()

#drug.GDSC %>% head()

drug.GDSC <- readxl::read_excel('./../data/GDSC1_fitted_dose_response_15Oct19.xlsx')
#drug.GDSC <- GDSC.targets %>% select(drug_name, target) %>% unique() %>% merge(drug.GDSC, ., by.x='drug', by.y = 'drug_name', all.x=T)

GDSC.meta <- readxl::read_excel('./../data/Cell_Lines_Details.xlsx') %>% merge(select(meta.dat, DepMap_ID, `CCLE Name`, COSMIC_ID), by.x='COSMIC identifier', by.y='COSMIC_ID') 

#GDSC.meta %>% head()

GDSC.dat <- drug.GDSC %>% merge(select(GDSC.meta, DepMap_ID, `Sample Name`), by.x='CELL_LINE_NAME', by.y='Sample Name') %>% select(DepMap_ID, DRUG_NAME, PUTATIVE_TARGET, AUC) %>% mutate(drug_name=DRUG_NAME, target=PUTATIVE_TARGET, response=AUC, response_type='AUC_GDSC') %>% select(DepMap_ID, drug_name, target, response, response_type) %>% mutate(id=DepMap_ID, id_type='DepMap_ID') %>% drop_na() %>% select(-DepMap_ID)

GDSC.dat %>% head()
```

# Drug Response (CTRP CTD²)

CTD²: Cancer Target Discovery And Development

## Sources  

[paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4631646/#SD1)  

#### Supplemental 4  
[data source](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4631646/bin/NIHMS711523-supplement-4.xlsx)  

## References 

Harnessing Connectivity in a Large-Scale Small-Molecule Sensitivity Dataset Seashore-Ludlow et al., Cancer Discovery, 5, 1210-1223 (2015) and Correlating chemical sensitivity and basal gene expression reveals mechanism of action Rees et al., Nat Chem Biol, 12, 109-116 (2016)  

Include:   
"The results published here are in whole or part based upon data generated by Cancer Target Discovery and Development (CTD2) Network (https://ocg.cancer.gov/programs/ctd2/data-portal) established by the National Cancer Institute's Office of Cancer Genomics."  

Aksoy, Dancík, Smith et al., Database 2017;1-10 and provide the URL https://ctd2-dashboard.nci.nih.gov/dashboard/.  

```{r}
# get everything from supplemental data 
CTRP.resp <- readxl::read_excel('./../data/NIHMS711523-supplement-4.xlsx', sheet='S3')
#CTRP.resp %>% head()

CTRP.ccl <- readxl::read_excel('./../data/NIHMS711523-supplement-4.xlsx', sheet='S2')
#CTRP.ccl %>% head()

CTRP.cpd <- readxl::read_excel('./../data/NIHMS711523-supplement-4.xlsx', sheet='S1')
#CTRP.cpd %>% head()

CTRP.dat <- CTRP.resp %>% merge(CTRP.ccl, by='index_ccl') %>% merge(CTRP.cpd, by='index_cpd') %>% merge(meta.dat, by.x='cell_line_name', by.y='stripped_cell_line_name') %>% mutate(response=area_under_sensitivity_curve, response_type='CTRP_AUC', target=gene_name_of_protein_target,drug_name=compound_name) %>% select(DepMap_ID, drug_name, target, response, response_type) %>% mutate(id=DepMap_ID, id_type='DepMap_ID') %>% select(-DepMap_ID) %>% drop_na()

CTRP.dat %>% head()
```
```{r}
CTRP.dat$DepMap_ID %>% unique() %>% length()
CTRP.dat$drug_name %>% unique() %>% length()
```

# Drug Response (CCLE NP24)

Pharmacologic profiles for 24 anticancer drugs across 504 cell lines.

### Act Area = AUC Area 

> Additionally, we calculated two forms of the Activity area for each curve,
defined as the area between the response curve and a fixed reference Aref = 0 or a variable
reference Aref = max(0, Alow) where Alow is the activity at the lowest concentration, up to
the maximum tested concentration. In practice, the Activity area was calculated as the
sum of differences between the measured Ai at concentration i and the reference level.
Thus, using the fixed reference, Activity area = 0 corresponds to an inactive compound,
and 8 corresponds to a compound which had A = -100% at all eight concentrations
points. The variable reference form was introduced to adjust for curves with large
positive activities close to zero concentration, which are usually artifacts of imperfectly
corrected variations on the assay plate. For this measure, the median of all replicate
activity values was used regardless of cell line run day. To prevent confusion, the Activity
Area was calculated using Aref = 0 unless otherwise noted. [supplementary, pg11](https://media.nature.com/original/nature-assets/nature/journal/v483/n7391/extref/nature11003-s2.pdf)  

## Sources  

[more info](https://depmap.org/portal/ccle/)

[source- drug data](https://depmap.org/portal/download/api/download/external?file_name=ccle_legacy_data%2Fpharmacological_profiling%2FCCLE_NP24.2009_Drug_data_2015.02.24.csv)

[source- profile data](https://depmap.org/portal/download/api/download/external?file_name=ccle_legacy_data%2Fpharmacological_profiling%2FCCLE_NP24.2009_profiling_2012.02.20.csv)

[source- activity data](https://depmap.org/portal/download/api/download/external?file_name=ccle_legacy_data%2Fpharmacological_profiling%2FCCLE_GNF_data_090613.xls)

## References 

Cancer Cell Line Encyclopedia Consortium, and Genomics of Drug Sensitivity in Cancer Consortium. 2015. Pharmacogenomic Agreement between Two Cancer Cell Line Data Sets. Nature 528 (7580):84-87. https://doi.org/10.1038/nature15736.  

Jordi Barretina, Giordano Caponigro, Nicolas Stransky, Kavitha Venkatesan, William R. Sellers, Robert Schlegel, Levi A. Garraway, et. al. 2012. The Cancer Cell Line Encyclopedia Enables Predictive Modelling of Anticancer Drug Sensitivity. Nature 483 (7391):603-7. https://doi.org/10.1038/nature11003.  

```{r}
drug.CCLE <- as.data.frame(fread('./../data/CCLE_NP24.2009_Drug_data_2015.02.24.csv', sep=','))
#drug.CCLE %>% head()

CCLE.profile <- as.data.frame(fread('./../data/CCLE_NP24.2009_profiling_2012.02.20.csv', sep=','))
#CCLE.profile %>% head()

CCLE.activity <-readxl::read_excel('./../data/CCLE_GNF_data_090613.xls')
#CCLE.activity %>% head()

# relevant features 
CCLE.dat <- drug.CCLE %>% select(`CCLE Cell Line Name`, Compound, Target, ActArea) %>% merge(unique(select(meta.dat, DepMap_ID, `CCLE Name`)), by.x='CCLE Cell Line Name', by.y='CCLE Name') %>% mutate(response=ActArea, target=Target, drug_name=Compound, response_type='AUC_drug_CCLE') %>% select(DepMap_ID,drug_name, target, response, response_type) %>% mutate(id=DepMap_ID, id_type='DepMap_ID') %>% drop_na() %>% select(-DepMap_ID)

CCLE.dat %>% head()

```

# beatAML (waves 1,2) Drug Response Data

122 single agent drugs  

528 patients  

## Sources 

[source-ctd2](https://ctd2-data.nci.nih.gov/Public/OHSU-1/BeatAML_Waves1_2/)

## References 

Please credit the NCI Genomic Data Commons (GDC) in your manuscript by citing the following paper about the GDC in your manuscript:

  Grossman, Robert L., Heath, Allison P., Ferretti, Vincent, Varmus, Harold E., Lowy, Douglas R., Kibbe, Warren A., Staudt, Louis M. (2016) Toward a Shared Vision for Cancer Genomic Data. New England Journal of Medicine375:12, 1109-1112  
  
Functional genomic landscape of acute myeloid leukaemia | Nature. https://www.nature.com/articles/s41586-018-0623-z.  

```{r}
beatAML.drug <- as.data.frame(fread('./../data/beatAML_waves1_2/OHSU_BeatAMLWaves1_2_Tyner_DrugResponse.txt', sep='\t')) %>% mutate(drug_name=inhibitor, response=auc, response_type='beatAML_AUC') %>% select(lab_id, drug_name, response, response_type)
beatAML.drug %>% head()

#beatAML.drug$drug_name %>% unique() %>% length() #122
#beatAML.drug$lab_id %>% unique() %>% length() #528
```

# beatAML Expression Data - FPKM

Still need to convert Ensembl gene IDs to gene name (that depmap uses) and filter to set union on DepMap expr data

**NOTE:** beatAML expression data is in log2 RPKM, but DepMap expression data is in TPM.   

## Data Access & Processing   

Use `RPKM` workflow data types: 510 files. To download:     
1. Make a manifest file from the files listed in the `source` link below    
2. download the gdc-client tool ([info](https://docs.gdc.cancer.gov/Data_Transfer_Tool/Users_Guide/Getting_Started/))  
3. Download manifest listed files by running: 
        ```bash $ gdc-client download -m ./../beatAML-EXPR-manifest_RPKM.txt```    
4. Extract, normalize RPKM (log2, pseudocount of 1) and combine data by running (in the python dir):   
        ```bash $ python extract_beatAML_expr_RPKM.py ./../data/BeatAML_Waves1_2/EXPR_DATA_RPKM/ ./../data/beatAML_EXPR_RPKM.csv  ```
        
[custom normalization](https://www.biostars.org/p/218995/):: use unstranded read count and convert to z-score 
[for future - expression normalization techniques](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0206312)
What normalization method does DepMap use? TPM.    

## sources   

Make [manifest](https://portal.gdc.cancer.gov/repository?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22content%22%3A%7B%22field%22%3A%22cases.case_id%22%2C%22value%22%3A%5B%22set_id%3AAW43dUlkTt_rMbGdDakD%22%5D%7D%2C%22op%22%3A%22IN%22%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.analysis.workflow_type%22%2C%22value%22%3A%5B%22HTSeq%20-%20FPKM%22%5D%7D%7D%2C%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_type%22%2C%22value%22%3A%5B%22Gene%20Expression%20Quantification%22%5D%7D%7D%5D%7D)  

[NCI GDC beatAML project page](https://portal.gdc.cancer.gov/projects/BEATAML1.0-COHORT)  

[source-meta](http://vizome.org/aml/)  


```{r}
beatAML.meta <- as.data.frame(fread('./../data/BeatAML_Waves1_2/All_sample_data.csv', sep=','))
#beatAML.meta %>% head()

expr.temp<-as.data.frame(fread('./../data/beatAML_EXPR_FPKM.csv', sep=','))
genes <- colnames(expr.temp)[4:length(colnames(expr.temp))]

beatAML.expr <- as.data.frame(fread('./../data/beatAML_EXPR_FPKM.csv', sep=',')) %>% mutate(entity_id=substr(entity_id, 1, nchar(entity_id)-1)) %>% merge(select(beatAML.meta, SampleID, LLS_SampleID), by.x='entity_id', by.y='SampleID') %>% mutate(lab_id=LLS_SampleID) %>% select(lab_id, genes) 

beatAML.expr %>% head()

```

```{r}

beatAML.expr <- beatAML.expr %>% gather(-lab_id, key='ensembl_id', value='expression') %>% separate(ensembl_id, into=c('ensembl_id', 'ensembl_id_version')) %>% merge(map2, by='ensembl_id', all.x=T)

beatAML.expr %>% head()

```

```{r}
beatAML.expr %>% write_delim('./../data/processed/beataml_expr.csv', delim=',')
```


# Targetome 

Need to use this to link beatAML targets  

160 Unique Drugs with targets  

--- 
### From Rory 
---

#### targetome_DrugInformation_070617.txt  
- drugs by class/categories (ATC, EPC, etc) + FDA approval date   

#### targetome_DrugSynonymDictionary_100917.txt  
- mapping for drug->synonyms    
- includes chemical, brand, generic, proprietary, investigational names as available    
- note IMO it's best to use the full synonym dictionary because there is poor consistency about drug name usage :/   

#### targetome_Evidence_TIPS_101017.txt  
- includes target evidence for ~140 drugs    

#### targetome_FullEvidence_011019.txt  
- includes target evidence for ~170 drugs    
- this is all drugs we originally mined, note we subsetted to anti-neoplastic (ATC class with prefix "L01" for cancer targetome paper)    
- +small bug fix, there were a few rows with blank assay values due to a range of nM values being reported, added "range" flag to indicate    

---

Drug targets - [more info](https://www.ncbi.nlm.nih.gov/pubmed/28964549)  

#### References 

Evidence-Based Precision Oncology with the Cancer Targetome. - PubMed - NCBI.   https://www.ncbi.nlm.nih.gov/pubmed/28964549.  

```{r}

targetome.druginfo <- as.data.frame(fread('./../data/targetome/Targetome_DrugInformation_070617.txt', sep='\t', header=T))
targetome.drugsynonym <- as.data.frame(fread('./../data/targetome/Targetome_DrugSynonymDictionary_100917.txt', sep='\t', header=T))
targetome.evidence.tips <- as.data.frame(fread('./../data/targetome/Targetome_Evidence_TIPS_101017.txt', sep='\t'))
targetome.evidence.full <- as.data.frame(fread('./../data/targetome/Targetome_FullEvidence_011019.txt', sep='\t'))

beatAML.targetome.overlap <- as.data.frame(fread('./../data/targetome/Inhibitors_inCancerTargetome_110217.txt', sep='\t'))

targetome.druginfo %>% head() 
targetome.drugsynonym %>% head()
targetome.evidence.tips %>% head()
targetome.evidence.full %>% head()
beatAML.targetome.overlap %>% head()

#targetome.evidence.full$Drug %>% unique() %>% length() # 160

# %>% select(Drug) %>% unique() %>% arrange(Drug) 
```


## beatAML link targets 

combine targets from targetome (32 drugs) to our beatAML data. 

```{r}

# which tiers do we want to include in our targets 
TIERS = c('III')

beatAML.dat <- beatAML.drug %>% merge(beatAML.targetome.overlap, by.x='drug_name', by.y='CancerTargetome_Drug') %>%
  merge(targetome.evidence.full, by.x='InhibitorName_Panel', by.y='Drug') %>% 
  filter(EvidenceLevel_Assigned %in% TIERS) %>% 
  select(drug_name, lab_id, response, response_type, Target_Name) %>% 
  unique() %>% 
  aggregate(Target_Name ~ drug_name + lab_id + response + response_type, data = ., paste, collapse = ";") %>% 
  mutate(target = Target_Name, id=lab_id, id_type='lab_id') %>% select(-Target_Name, -lab_id)

beatAML.dat %>% head()
```

## Combine Data 

```{r}
all.drug <- rbind(rnai.dat, crispr.dat, PRISM.dat, GDSC.dat, CTRP.dat, CCLE.dat, beatAML.dat)

write_delim(all.drug, './../data/processed/all_resp_data.csv', delim=',', row.names=F)
```








